controller/node_controller.go

```diff
@@ -687,6 +690,111 @@ func (nc *NodeController) syncDiskStatus(node *longhorn.Node, collectedDataInfo
 	return nc.updateDiskStatusSchedulableCondition(node)
 }
 
+func (nc *NodeController) syncDiskSchedule(node *longhorn.Node) error {
+	if err := nc.cleanupDiskSchedule(node); err != nil {
+		return err
+	}
+
+	if err := nc.syncDiskScheduleSpec(node); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (nc *NodeController) cleanupDiskSchedule(node *longhorn.Node) error {
+	nodeDisks := map[string]bool{}
+	if node.DeletionTimestamp.IsZero() && node.Status.DiskStatus != nil {
+		for _, diskStatus := range node.Status.DiskStatus {
+			nodeDisks[diskStatus.DiskUUID] = true
+		}
+	}
+
+	diskScheduleMap, err := nc.ds.ListDiskSchedulesOnNode(node.Name)
+	if err != nil && !datastore.ErrorIsNotFound(err) {
+		return errors.Wrapf(err, "failed to list disk schedule on node %v for clean up", node.Name)
+	}
+
+	deleteErrs := multierr.NewMultiError()
+	for _, diskSchedule := range diskScheduleMap {
+		if !nodeDisks[diskSchedule.Name] {
+			nc.logger.Infof("Clearning disk schedule %v (%q) on node %v", diskSchedule.Name, diskSchedule.Spec.Name, node.Name)
+			if deleteErr := nc.ds.DeleteDiskSchedule(diskSchedule.Name); deleteErr != nil {
```

Comment:
nit:
Simplify the variable naming. Apply to other functions.

```suggestion
			if err := nc.ds.DeleteDiskSchedule(diskSchedule.Name); err != nil {
```